### PR TITLE
LibGL+LibSoftGPU+LibGfx: Improvements for Tux Racer, Grim Fandango

### PR DESCRIPTION
**Pull request by GMTA**
Visual improvements:
- Mipmap selection in `Sampler` changed so Grim Fandango subtitles are rendered correctly
- Normal transformation in `Device` fixed so Tux in Tux Racer is lit correctly

Grim Fandango before:
Screenshot_20220315_102003

Grim Fandango after:
Screenshot_20220315_102214

Tux Racer before:
image

Tux Racer after:
image

### CODE REVIEW

File: Userland/Libraries/LibGfx/Matrix4x4.h [hunk 1]
```diff
@@ -37,16 +37,6 @@ constexpr static Vector3<T> transform_point(const Matrix4x4<T>& m, const Vector3
         p.x() * elements[2][0] + p.y() * elements[2][1] + p.z() * elements[2][2] + elements[2][3]);
 }
 
-template<typename T>
```
Comment (sunverwerth) [inline Userland/Libraries/LibGfx/Matrix4x4.h:40]: Why was this removed?

Comment (GMTA) [reply Userland/Libraries/LibGfx/Matrix4x4.h:40]: Its only usage was removed, and I'd imagine this function was very OpenGL-specific. If you think there is a good remaining use case I'll leave it in.

Comment (GMTA) [reply Userland/Libraries/LibGfx/Matrix4x4.h:40]: Also, I don't think the original code was correct? It transforms the normals with the model view transform, but we should apply the inverse transformation instead. That is what `normal_transform` is for.

Comment (sunverwerth) [reply Userland/Libraries/LibGfx/Matrix4x4.h:40]: It the counterpart to `transform_point()`.
transform_point uses an implied w coordinate of 1 and transform_direction uses an implied w coordinate of 0. Very handy when you want to multiply vec3s by a 4x4 matrix which would otherwise require constructing a vec4 first.

Comment (GMTA) [reply Userland/Libraries/LibGfx/Matrix4x4.h:40]: As discussed on Discord; we believe that `transform_direction` is a too specialized case to keep. I've tried to generalize things a bit by adding `Matrix::submatrix_from_top_left<N>()`. Let me know what you think!

Comment (sunverwerth) [reply Userland/Libraries/LibGfx/Matrix4x4.h:40]: Yeah looks good. Maybe add a fixme to transform_point if you son't want to remove this right now.

File: Userland/Libraries/LibGfx/Matrix4x4.h [hunk 2]
```diff
@@ -37,16 +37,6 @@ constexpr static Vector3<T> transform_point(const Matrix4x4<T>& m, const Vector3
         p.x() * elements[2][0] + p.y() * elements[2][1] + p.z() * elements[2][2] + elements[2][3]);
```
Comment (sunverwerth) [inline Userland/Libraries/LibGfx/Matrix4x4.h:37]: @GMTA not sure if you noticed my last message, so here it is again: I think it would be good to add a fixme here if you are not going to remove `transform_point` right away.

Comment (GMTA) [reply Userland/Libraries/LibGfx/Matrix4x4.h:39]: @sunverwerth Thanks for reminding me; is this what you meant?

Comment (sunverwerth) [reply Userland/Libraries/LibGfx/Matrix4x4.h:39]: Yes

**Review by sunverwerth — APPROVED** [review]

**Review by Quaker762 — APPROVED** [review]
Looks good to me! LGTM!
